Purchases/Public/CustomerInfo.swift

```diff
@@ -163,11 +163,11 @@ import Foundation
 
     convenience init(data: [String: Any]) throws {
         try self.init(data: data,
-                      dateFormatter: .iso8601SecondsDateFormatter,
+                      dateFormatter: ISO8601DateFormatter.default,
                       transactionsFactory: TransactionsFactory())
     }
 
-    init(data: [String: Any], dateFormatter: DateFormatter, transactionsFactory: TransactionsFactory) throws {
+    init(data: [String: Any], dateFormatter: DateFormatterType, transactionsFactory: TransactionsFactory) throws {
```

Comment:
I feel like using our `DateFormatterType` instead of the `Foundation` `DateFormatter` might make our API slightly more confusing to use. I'm definitely open to the change, I just want to get folks thoughts on that aspect. Would it make sense to keep the old `init` args the same?

Reply:
the init isn't public, is it?

Reply:
`DateFormatter` implements the new `DateFormatterType`, so you can still pass a `DateFormatter`.
Additionally, as @beylmk pointed out, no public API references the new type, so I made it `internal`, and this is now an implementation detail.

We can't keep this as `DateFormatter` because unfortunately `ISO8601DateFormatter` inherits `Formatter`, not `DateFormatter`.

Reply:
🤦‍♂️ thank you for the info